cleanlab/rank.py

```diff
@@ -652,3 +653,106 @@ def get_outlier_scores(
+    that are less typical under the dataset distribution (values near 0 indicate outliers).
+    TODO: what is the score based on.
+
+    Parameters
+    ----------
+    pred_probs : np.ndarray
+      TODO
+
+    confident_thresholds : np.ndarray, default = None
+      TODO
+    labels : np.ndarray, default = None
+      TODO
+
+    adjusted_pred_probs : bool, default = False
+      TODO
+
+    scoring_method : bool, default = False
+      TODO
+
+    t : int, default=1
+      Optional hyperparameter only for advanced users.
+      Controls transformation of distances between examples into similarity scores that lie in [0,1].
+      The transformation applied to distances `x` is `exp(-x*t)`.
+      If you find your scores are all too close to 1, consider increasing `t`,
+      although the relative scores of examples will still have the same ranking across the dataset.
+
+    return_thresholds : bool, default = False
+      Whether the `confident_thresholds` array should also be returned.
+      If True, this function returns a tuple `(ood_scores, confident_thresholds)`.
+    Returns
+    -------
+    ood_scores : np.ndarray
+      Score for each example that roughly corresponds to the likelihood this example stems from the same distribution as
+      the dataset features (i.e. is not an outlier).
+      If ``return_thresholds = True``, then a tuple is returned
+      whose first element is array of `ood_scores` and second is an np.ndarray of `confident_thresholds`.
+    """
+
+    valid_methods = [
+        "entropy",
+        "self_confidence",
+    ]
+
+    # Might be a better place to put this warning
+    if (confident_thresholds is not None or labels is not None) and not adjusted_pred_probs:
+        warnings.warn(
+            f"OOD scores are not adjusted with confident thresholds. If scores need to be adjusted set adjusted_pred_probs = True. Otherwise passing in confident_thresholds and/or labels does not change score calculation.",
+            UserWarning,
+        )
+
+    # TODO warn about both thresholds and labels passed in
+
+    if confident_thresholds is None:
+        if adjusted_pred_probs:
+            if labels is None:
+                raise ValueError(
+                    f"Cannot calculate adjusted_pred_probs without labels. Either pass in labels parameter or set adjusted_pred_probs = False."
+                )
+            confident_thresholds = get_confident_thresholds(
+                labels, pred_probs, multi_label=False
+            )  # should we support multi_label?
+        else:
+            confident_thresholds = 0
+
+    if adjusted_pred_probs:
+        pred_probs = pred_probs - confident_thresholds
+        pred_probs += confident_thresholds.max()
+        pred_probs /= pred_probs.sum(axis=1)[:, None]
+
+    if scoring_method == "entropy":
+        ood_scores = get_normalized_entropy(pred_probs)
+    elif scoring_method == "self_confidence":
```

Comment:
Let's change this method's name to: `"least_confidence"` to match our label-quality scoring paper and the uncertainty sampling literature.  Note it is slightly different than our self_confidence which measures the probability of the given label, not the probability of the argmax_class.